omero/developers/json-api.txt

```diff
@@ -0,0 +1,460 @@
+JSON API
+========
+
+Overview
+--------
+
+The OMERO json API described here provides create, read, update and delete
+access to an underlying OMERO server.
+
+
+Omero-marshal and Projection-based APIs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The majority of the API urls use `omero-marshal <https://github.com/openmicroscopy/omero-marshal>`_
+to generate json dictionaries from OMERO **model** objects. The json dictionaries will be populated
+with all fields of the OMERO model object, such that changing a single field and saving this
+back to OMERO will update the object accordingly.
+All these urls are under the ``m`` prefix:
+
+::
+
+    <server>/api/v0/m/
+
+
+The webclient currently uses a small number of urls to perform customised
+queries for browsing Project, Dataset & Image hierarchies.
+These queries use **projections** and typically load a subset of fields for
+OMERO objects in order to improve performance for large data counts.
+These will be made available under the ``p`` prefix in future releases
+but are not yet supported in OMERO 5.3.0.
+
+::
+
+    <server>/api/v0/p/
+
+
+Versioning
+^^^^^^^^^^
+The JSON API uses major and minor version numbers to reflect breaking
+and non-breaking changes respectively. Non-breaking changes include simple
+addition of attributes to JSON data or addition of new urls.
+The API version is not tied to the version of OMERO.server.
+
+The major version is included in the url ``v0`` whereas the full
+version number can be found in the header:
+
+::
+
+    X-OMERO-ApiVersion: 0.0
+
+
+Pagination
+^^^^^^^^^^
+Requests that return a list of items will be paginated, showing
+a ``limit`` of the first 200 objects by default.
+Pagination can be specified using the ``limit`` and ``offset`` query
+parameters:
+
+::
+
+    # List the first 100 objects (offset=0 by default)
+    <server>/api/v0/m/projects/?limit=100
+
+    # List the next 100 objects
+    <server>/api/v0/m/projects/?limit=100&offset=100
+
+
+Pagination details will be returned in a ``meta`` json object, including
+the ``totalCount`` of objects for that query, the current ``offset`` and
+``limit`` as well as the ``maxLimit`` that you can use.
+
+::
+
+    "meta": {
+        "totalCount": 13240,
+        "maxLimit": 500,
+        "limit": 200,
+        "offset": 0
+    },
+
+Sysadmins can configure the default ```limit``` and ```maxLimit``` settings
+for their server, for example:
+
+::
+
+    $ bin/omero config set omero.web.api.limit 100
+    $ bin/omero config set omero.web.api.max_limit 300
+
+
+Normalizing Experimenters and Groups
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When returning a list of JSON objects that each contain ```omero:details``` with
```

Comment:
All the mark up in this section has an extra set ` that is rendered in the staging pages too